src/simulation/test/LoadGeneratorTests.cpp

```diff
@@ -475,6 +480,90 @@ TEST_CASE("generate soroban load", "[loadgen][soroban]")
             keys.insert(lk);
         }
     }
+
+    // Test MIXED_CLASSIC_SOROBAN mode
+    SECTION("Mix with classic")
+    {
+        constexpr uint32_t numMixedTxs = 200;
+        auto mixLoadCfg = GeneratedLoadConfig::txLoad(
+            LoadGenMode::MIXED_CLASSIC_SOROBAN, nAccounts, numMixedTxs,
+            /* txRate */ 1);
+
+        auto& mixCfg = mixLoadCfg.getMutMixClassicSorobanConfig();
+        mixCfg.payWeight = 50;
+        mixCfg.sorobanInvokeWeight = 45;
+        constexpr uint32_t uploadWeight = 5;
+        mixCfg.sorobanUploadWeight = uploadWeight;
+
+        // Because we can't preflight TXs, some invocations will fail due to too
+        // few resources. This is expected, as our instruction counts are
+        // approximations. Additionally, many upload transactions will fail as
+        // they are likely to generate invalid wasm. Therefore, we check that
+        // all but `maxInvokeFail + 1.5 * uploadWeight` transactions succeed. In
+        // case the random sampling produces more upload transactions than
+        // expected, we allow for a 50% margin of error on the number of upload
+        // transactions.
+        constexpr int maxSorobanFail = 1.5 * uploadWeight + maxInvokeFail;
+        mixLoadCfg.setMinSorobanPercentSuccess(100 - maxSorobanFail);
+
+        loadGen.generateLoad(mixLoadCfg);
+        auto numSuccessBefore = getSuccessfulTxCount();
+        auto numFailedBefore =
+            app.getMetrics()
+                .NewCounter({"ledger", "apply-soroban", "failure"})
+                .count();
+        simulation->crankUntil(
+            [&]() {
+                return app.getMetrics()
+                           .NewMeter({"loadgen", "run", "complete"}, "run")
+                           .count() == 6;
+            },
+            300 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+        // Check results
+        for (auto node : nodes)
+        {
+            // All classic transactions should succeed
+            auto& classicSucceeded =
+                node->getMetrics().NewCounter({"ledger", "apply", "success"});
+            auto& classicFailed =
+                node->getMetrics().NewCounter({"ledger", "apply", "failure"});
+            REQUIRE(classicFailed.count() == 0);
```

Comment:
I think we need to add an assert here to check that  `classicSucceeded` is within some allowed bounds. During loadgen, a TX could succeed or fail, but also be dropped. I think it may be possible for this test to pass even if many classic TXs are dropped and don't actually succeed.

Reply:
Good point. I added an assert in 52a7ca14257d86386d24283a380128d280ca625d